packages/common/src/search/Collection.ts

```diff
@@ -95,7 +95,7 @@ export class Collection implements IHubCollection {
         ],
       };
     } else {
-      qry = query;
+      qry = cloneObject(query);
```

Comment:
This is the actual "fix" in this PR. Without this, the IQuery accumulated filters as the Collections were iterated